arrows/kpf/yaml/kpf_canonical_types.h

```diff
@@ -76,8 +76,8 @@ struct KPF_YAML_EXPORT bbox_t
 struct KPF_YAML_EXPORT id_t
 {
   enum {DETECTION_ID=0, TRACK_ID, EVENT_ID };
-  size_t d;
-  explicit id_t( size_t i ): d(i) {}
+  unsigned long long d;
```

Comment:
`unsigned long long` sounds like an implementation dependent type. Would it be better to use `uint64_t`?
(Other files too)

Reply:
Originally the actual change introduced by this branch are relatively minor (I don't believe it touched this file at all), so a vast majority of the changes are from rebases on other branches to ensure that this branch is ready to merge when those land.

Reply:
This may be one of the changes that the dev/overhaul-final-rebase branch fixes. That branch should be merging into master soon. After it does, I'll rebase this branch to be on top of master, then some of these issues may just resolve themselves.